lib/cli/cli.js

```diff
@@ -32,7 +29,33 @@ exports.main = (argv = process.argv.slice(2)) => {
 
   Error.stackTraceLimit = Infinity; // configurable via --stack-trace-limit?
 
-  yargs
+  const args = exports.parseArgv(argv);
+  const {handler} = require(`./${args.command.name}`);
+  args.command = null;
+  handler(args);
+};
+
+exports.yargsParse = function(argv) {
+  const {loadOptions} = require('./options');
+  return exports
+    .createYargs()
+    .parse(argv, Object.assign(loadOptions(argv), {command: {}}));
+};
```

Comment:
I haven't understood yet (neither in the current master) why the object returned by `loadOptions()` is passed as `context` parameter to `parse()`.
- the coerce functions are not called that way, see #3826 
- duplicate alias option don't get eliminated

Reply:
Yeah, sounds like a separate bug.

Reply:
Even `node` flags are allowed in config files. When run with `_mocha` they are useless, I don't think they get filtered out, probably they are just ignored.